vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1144,6 +1167,12 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	log.Printf("[DEBUG] Device list %+v", deviceList)
 	for _, device := range deviceList {
 		networkInterface := make(map[string]interface{})
+		adapter_type_string := fmt.Sprintf("%T", device)
```

Comment:
Here, rather than printing the type out, you should be using the more spec-native [type assertion](https://tour.golang.org/methods/15).

The code below would look like this:

```go
if _, ok := device.(*types.VirtualE1000); ok {
  networkInterface["adapter_type"] = "e1000"
} else {
  networkInterface["adapter_type"] = "vmxnet3"
}
``` 

You can also use the [`switch`](https://tour.golang.org/methods/16) type assertion format as well.